src/eins-hwinfo.c

```diff
@@ -105,159 +115,70 @@ round_to_nearest (guint64 size,
   return (guint32) CLAMP (size / divisor, 0, G_MAXUINT32);
 }
 
-GVariant *
-eins_hwinfo_get_disk_space_for_partition (GFile   *file,
-                                          GError **error)
+guint32
+eins_hwinfo_get_ram_size (void)
+{
+  glibtop_mem mem;
+  guint32 size_mib;
+
+  glibtop_get_mem (&mem);
+  size_mib = round_to_nearest (mem.total, ONE_MIB_IN_BYTES);
+  return size_mib;
+}
+
+gboolean
+eins_hwinfo_get_disk_space_for_partition (GFile          *file,
+                                          DiskSpaceType  *diskspace)
 {
   g_autoptr(GFileInfo) info = NULL;
+  g_autoptr(GError) error = NULL;
   guint64 total = 0;
   guint64 used = 0;
-  guint64 available = 0;
+  guint64 free = 0;
+
+  if (diskspace == NULL)
+    {
+      g_error ("Wrong memory address to hold disk space information");
+      return FALSE;
+    }
```

Comment:
If you want to abort the process in this case (which `g_error` will do), just use:

```suggestion
  g_assert (diskspace != NULL);
```

If you do not want to abort the process, use:

```
  g_return_val_if_fail (diskspace != NULL, FALSE);
```

Reply:
Thanks for the hint.  I did not know that special behavior of [g_error()](https://developer.gnome.org/glib/stable/glib-Message-Logging.html#g-error) and thought it is just log level, until this time. :)